odbc/src/api/functions.rs

```diff
@@ -1785,11 +1785,441 @@ pub unsafe extern "C" fn SQLGetInfo(
+        | InfoType::ConvertGuid => {
+            // Binary, Date, Time, VarBinary, LongVarBinary, and GUID
+            // are invalid CAST source types.
+            return i16_len::set_output_fixed_data(&SQL_U32_ZERO, info_value_ptr, string_length_ptr);
+        }
+        // SQL_GETDATA_EXTENSIONS
+        InfoType::GetDataExtensions => {}
+        // SQL_COLUMN_ALIAS
+        InfoType::ColumnAlias => {
+            // MongoSQL does support column aliases.
+            return i16_len::set_output_wstring(
+                SQL_INFO_Y,
+                info_value_ptr as *mut WChar,
+                buffer_length as usize,
+                string_length_ptr,
+            );
+        }
+        // SQL_GROUP_BY
+        InfoType::GroupBy => {}
+        // SQL_ORDER_BY_COLUMNS_IN_SELECT
+        InfoType::OrderByColumnsInSelect => {
+            // MongoSQL does require ORDER BY columns to be in the SELECT list.
+            return i16_len::set_output_wstring(
+                SQL_INFO_Y,
+                info_value_ptr as *mut WChar,
+                buffer_length as usize,
+                string_length_ptr,
+            );
+        }
+        // SQL_OWNER_USAGE
+        InfoType::OwnerUsage => {
+            // SQL_OWNER_USAGE is replaced by SQL_SCHEMA_USAGE in newer
+            // ODBC versions. They use the same numeric value.
+            //
+            // MongoSQL does not have a concept of a "schema" which is a
+            // level above "catalog" (or "database") in the data hierarchy.
+            return i16_len::set_output_fixed_data(&SQL_U32_ZERO, info_value_ptr, string_length_ptr);
+        }
+        // SQL_CATALOG_USAGE
+        InfoType::CatalogUsage => {
+            // TODO: (code review) there is no way to indicate this is read-only. The entire DML bitmask
+            //       must be used to indicate we support SELECT. However, that also includes INSERT,
+            //       UPDATE, and DELETE.
+            return i16_len::set_output_fixed_data(&SQL_CU_DML_STATEMENTS, info_value_ptr, string_length_ptr);
+        }
+        // SQL_SPECIAL_CHARACTERS
+        InfoType::SpecialCharacters => {
+            // TODO: (code review) The docs say:
+            //  - "A character string that contains all special characters (that is,
+            //     all characters except a through z, A through Z, 0 through 9, and
+            //     underscore) that can be used in an identifier name, such as a table
+            //     name, column name, or index name, on the data source. For example,
+            //     "#$^". If an identifier contains one or more of these characters, the
+            //     identifier must be a delimited identifier."
+            //  - MongoSQL's grammar defines an identifier character as [^\x00], as in, literally any non-null character.
+            //    Surely we aren't meant to return a "special characters" string that contains every character, are we?
+        }
+        // SQL_MAX_COLUMNS_IN_GROUP_BY
```

Comment:
[nit] You could bundle all the MaxColumnsXXX info type cases since there is no max for any of them.

Reply:
done